dnSpy/dnSpy/Documents/Tabs/DefaultDocumentList.cs

```diff
@@ -48,6 +49,60 @@ sealed class DefaultDocumentList {
 		public void Add(DsDocumentInfo file) => files.Add(file);
 	}
 
+	sealed class DotnetReferenceFileFinder {
+		readonly CancellationToken cancellationToken;
+		readonly List<DefaultDocumentList> allLists;
+
+		public DotnetReferenceFileFinder(CancellationToken cancellationToken) {
+			this.cancellationToken = cancellationToken;
+			allLists = new List<DefaultDocumentList>();
+		}
+
+		public IEnumerable<DefaultDocumentList> AllFiles => allLists.ToArray();
+
+		public void Find() {
+			cancellationToken.ThrowIfCancellationRequested();
+
+			if (!TryGetRuntimeFolder(out var runtimesFolder)) {
+				return;
+			}
+
+			foreach (var appDir in Directory.EnumerateDirectories(runtimesFolder)) {
+				foreach (var versionDir in Directory.EnumerateDirectories(appDir)) {
+					var appName = $"{Path.GetFileName(appDir)} {Path.GetFileName(versionDir)}";
+					allLists.Add(new DefaultDocumentList(appName, Directory.GetFiles(versionDir, "*.dll").Where(FilterFiles).Select(DsDocumentInfo.CreateDocument)));
+				}
+			}
+		}
+
+		static bool FilterFiles(string file) {
+
+			var fileName = Path.GetFileName(file);
+
+			//Here, only DLLs that are obviously not .NET assemblies are filtered out; other files are not filtered, and should not be affected, so to speak.
+			if (fileName.StartsWith("api-ms-win")) {
+				return false;
+			}
+
+			return true;
+		}
```

Comment:
I think this check should be extended to include more of the more common native dll files like `clrjit.dll`, `coreclr.dll`, etc. to exclude as many native files from these lists as possible.